include/triton/Dialect/Triton/IR/TritonOps.td

```diff
@@ -506,7 +506,7 @@ def TT_CatOp : TT_Op<"cat", [Pure,
 }
 
 def TT_JoinOp : TT_Op<"join", [
-    Pure, SameTypeOperands]> {
```

Comment:
why? Those should be no-op

Reply:
I wonder if changing cat was enough?

Reply:
Yeah that might be it. I can test that out